### DISCUSSION TITLE
How do I replace command cy.type() with fill command without changing the existing code?

### DISCUSSION THREAD

**Discussion by jacksparrow414** [cypress-io/cypress#17541] [Questions and Help]
The **fill()** command form [Cypress-fill-command](https://docs.cypress.io/plugins/directory).
Our project want replace **cy.type()** with **fill()** command,but We don't want to change too much code.

Our idea is overwrite cy.type,then We want to integrate command cy.type and command fill into the overwrite command,which is controlled by an environment variable.

My command.js code is as follows
```javascript
import 'cypress-fill-command'
...

Cypress.Commands.overwrite('type', (originalFn, element, text, options) => {
  let typeDefault = Cypress.env("typeDefault");
  if (typeDefault) {
    Cypress.log({
      $el: element,
      name: 'type',
      message: 'type command',
    })
    return originalFn(element, text, options)
  }else {
    Cypress.log({
      $el: element,
      name: 'type',
      message: 'fill command',
    })
    return cy.get(element).fill(text)
  }
})
```
This worked  in parts of the scene
```javascript
cy.get("input[name='password']").type("111111")
```
But there are problems in the following scenarios
```javascript
cy.get("input[name='password']").clear().type("111111")
```
It will happen CypressError like this 
```xml
Cypress detected that you returned a promise from a command while also invoking one or more cy commands in that promise.

The command that returned the promise was:

  > cy.clear()

The cy command you invoked inside the promise was:

  > cy.get()
```
If I want to take a subject after clear(),What should I do? Or What do you think is the right way
Anybody? I'd appreciate it if you could help

_Originally posted by @jacksparrow414 in https://github.com/cypress-io/cypress/discussions/17534_

**Comment by jennifer-shehane** [comment]
Converting this to a discussion. As an open source project with a small maintainer team we have to focus our time on bugs and features in the product, which Issues are reserved for.

**Comment by jacksparrow414** [comment]
the reght way is [here](https://github.com/DanielFerrariR/cypress-fill-command/issues/10)
